// ngx-driver/esm2020/lib/driver/driver.component.mjs
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "./util/service/driver.service";
import * as i3 from "ngx-liburg-icon";
import * as i4 from "@angular/common";
import * as i5 from "ngx-liburg";
export class DriverComponent {
    constructor(_activate, _driverService, _spinnerService) {
        this._activate = _activate;
        this._driverService = _driverService;
        this._spinnerService = _spinnerService;
    }
    ngOnInit() {
        this._driverService.driverData.subscribe(drivers => {
            this.isLoading = this._spinnerService.isLoading;
            this.dataSourceDrivers = drivers.map((driver, index) => {
                const model = driver;
                return {
                    actions: this._actionTableListDriver(),
                    editable: false,
                    model: {
                        ...model,
                    },
                };
            });
        });
    }
    changePage(event) {
        this._driverService.getDataDriver({ items: event.pageSize, page: event.pageIndex + 1 });
    }
    newEntry() {
        let emptyEntry = {};
        emptyEntry.editable = true;
        emptyEntry.actions = this._actionTableListDriver();
        emptyEntry.model = {};
        this.dataSourceDrivers = [...this.dataSourceDrivers, emptyEntry];
        console.log(this.dataSourceDrivers);
    }
    _actionTableListDriver() {
        return [
            {
                iconClass: "fa_solid:gauge",
                classCss: "edit",
                method: (row) => {
                    row.editable = !row.editable;
                    if (!row.editable) {
                        this._driverService.patchDataDriver(row.model)
                            .subscribe();
                    }
                },
            },
        ];
    }
}
DriverComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.3", ngImport: i0, type: DriverComponent, deps: [{ token: i1.ActivatedRoute }, { token: i2.DriverService }, { token: i3.SpinnerService }], target: i0.ɵɵFactoryTarget.Component });
DriverComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.2.3", type: DriverComponent, selector: "lib-driver", ngImport: i0, template: "<div class=\"driver-table-wrapper\" *ngIf=\"isLoading | async\">\n  <elix-table\n    [dataSource]=\"dataSourceDrivers\"\n    [showPagination]=\"true\"\n    [lenghtPagination]=\"dataSourceDrivers.length\"\n    (onPaginationChange)=\"changePage($event)\"\n    [addedNewEntry]=\"true\"\n    (onAddEntry)=\"newEntry()\"\n  >\n    <elix-column-text\n      className=\"fullName\"\n      [editRow]=\"true\"\n      field=\"name\"\n      name=\"Name\"\n    ></elix-column-text>\n    <elix-column-text\n      className=\"driverLicense\"\n      field=\"description\"\n      name=\"Driver License\"\n    ></elix-column-text>\n    <elix-column-two-cases\n      className=\"license\"\n      [editRow]=\"true\"\n      field=\"license\"\n      name=\"License\"\n    ></elix-column-two-cases>\n    <elix-column-icon-action\n      className=\"action\"\n      name=\"ACTION FIELD\"\n      field=\"action\"\n    ></elix-column-icon-action>\n  </elix-table>\n</div>\n\n<div *ngIf=\"!(isLoading | async)\">\n  <lib-spinner></lib-spinner>\n</div>\n", styles: [".driver-table-wrapper{box-shadow:#00000026 0 15px 25px,#0000000d 0 5px 10px}\n"], dependencies: [{ kind: "directive", type: i4.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "component", type: i5.TableComponent, selector: "elix-table", inputs: ["dataSource", "extensible", "extandble$", "footerShow", "footerMessageClass", "newElementExtandble", "showPagination", "lenghtPagination", "paginationClass", "addedNewEntry"], outputs: ["onAddEntry", "onPaginationChange"] }, { kind: "component", type: i5.ColumnTextComponent, selector: "elix-column-text", outputs: ["onValueChanges"] }, { kind: "component", type: i5.ColumnIconActionComponent, selector: "elix-column-icon-action" }, { kind: "component", type: i5.ColumnTwoCasesComponent, selector: "elix-column-two-cases", outputs: ["onValueChanges"] }, { kind: "component", type: i3.SpinnerComponent, selector: "lib-spinner" }, { kind: "pipe", type: i4.AsyncPipe, name: "async" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.3", ngImport: i0, type: DriverComponent, decorators: [{
            type: Component,
            args: [{ selector: 'lib-driver', template: "<div class=\"driver-table-wrapper\" *ngIf=\"isLoading | async\">\n  <elix-table\n    [dataSource]=\"dataSourceDrivers\"\n    [showPagination]=\"true\"\n    [lenghtPagination]=\"dataSourceDrivers.length\"\n    (onPaginationChange)=\"changePage($event)\"\n    [addedNewEntry]=\"true\"\n    (onAddEntry)=\"newEntry()\"\n  >\n    <elix-column-text\n      className=\"fullName\"\n      [editRow]=\"true\"\n      field=\"name\"\n      name=\"Name\"\n    ></elix-column-text>\n    <elix-column-text\n      className=\"driverLicense\"\n      field=\"description\"\n      name=\"Driver License\"\n    ></elix-column-text>\n    <elix-column-two-cases\n      className=\"license\"\n      [editRow]=\"true\"\n      field=\"license\"\n      name=\"License\"\n    ></elix-column-two-cases>\n    <elix-column-icon-action\n      className=\"action\"\n      name=\"ACTION FIELD\"\n      field=\"action\"\n    ></elix-column-icon-action>\n  </elix-table>\n</div>\n\n<div *ngIf=\"!(isLoading | async)\">\n  <lib-spinner></lib-spinner>\n</div>\n", styles: [".driver-table-wrapper{box-shadow:#00000026 0 15px 25px,#0000000d 0 5px 10px}\n"] }]
        }], ctorParameters: function () { return [{ type: i1.ActivatedRoute }, { type: i2.DriverService }, { type: i3.SpinnerService }]; } });
