// ngx-driver/esm2020/lib/driver/driver-routing.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from "@angular/router";
import { DriverComponent } from "./driver.component";
import { DriversResolver } from "./util/service/drivers.resolver";
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
const routes = [
    {
        path: '',
        component: DriverComponent,
        resolve: { drivers: DriversResolver },
    },
];
export class DriverRoutingModule {
}
DriverRoutingModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.3", ngImport: i0, type: DriverRoutingModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
DriverRoutingModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.3", ngImport: i0, type: DriverRoutingModule, imports: [CommonModule, i1.RouterModule], exports: [RouterModule] });
DriverRoutingModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.3", ngImport: i0, type: DriverRoutingModule, imports: [CommonModule,
        RouterModule.forChild(routes), RouterModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.3", ngImport: i0, type: DriverRoutingModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        RouterModule.forChild(routes),
                    ],
                    exports: [
                        RouterModule,
                    ],
                }]
        }] });
