// ngx-driver/esm2020/lib/driver/util/service/driver.service.mjs
import { HttpHeaders } from "@angular/common/http";
import { Inject, Injectable } from '@angular/core';
import { Subject } from "rxjs";
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "./local-storage.service";
import * as i3 from "ngx-liburg-icon";
export class DriverService {
    constructor(_httpClient, _localStorage, _spinnerService, environment) {
        this._httpClient = _httpClient;
        this._localStorage = _localStorage;
        this._spinnerService = _spinnerService;
        this.environment = environment;
        this.header = new HttpHeaders;
        this.driverData = new Subject();
        let tokenFromLocalStorage = _localStorage.geItem('token');
        this.header = new HttpHeaders().set('Authorization', 'Bearer ' + tokenFromLocalStorage);
    }
    getDataDriver(requestData = { items: 10, page: 1 }) {
        this._httpClient.get(`${this.environment.api}/driver/description/${requestData.items}/${requestData.page}`, { headers: this.header })
            .subscribe(resp => {
            this._spinnerService.sendValue(true);
            this.driverData.next(resp);
        });
    }
    patchDataDriver(row) {
        return this._httpClient.patch(`${this.environment.api}/driver`, row, { headers: this.header });
    }
}
DriverService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.3", ngImport: i0, type: DriverService, deps: [{ token: i1.HttpClient }, { token: i2.LocalStorageService }, { token: i3.SpinnerService }, { token: 'env' }], target: i0.ɵɵFactoryTarget.Injectable });
DriverService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "14.2.3", ngImport: i0, type: DriverService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.3", ngImport: i0, type: DriverService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: i2.LocalStorageService }, { type: i3.SpinnerService }, { type: undefined, decorators: [{
                    type: Inject,
                    args: ['env']
                }] }]; } });
