// ngx-driver/esm2020/lib/driver/util/service/local-storage.service.mjs
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class LocalStorageService {
    constructor() { }
    geItem(item) {
        if (item !== undefined)
            return JSON.parse(localStorage.getItem(item));
        throw new Error("Put KEy");
    }
    getKey(index) {
        localStorage.key(index);
    }
    setItem(item, value) {
        localStorage.setItem(item, JSON.stringify(value));
    }
    removeItem(item) {
        localStorage.removeItem(item.toString());
    }
    removeAll() {
        localStorage.clear();
    }
}
LocalStorageService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.3", ngImport: i0, type: LocalStorageService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
LocalStorageService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "14.2.3", ngImport: i0, type: LocalStorageService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.3", ngImport: i0, type: LocalStorageService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return []; } });
