// ngx-liburg/esm2020/lib/table/table-material.module.mjs
import { LayoutModule } from '@angular/cdk/layout';
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { MatButtonModule } from '@angular/material/button';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatIconModule } from '@angular/material/icon';
import { MatInputModule } from '@angular/material/input';
import { MatPaginatorModule } from '@angular/material/paginator';
import { MatSelectModule } from '@angular/material/select';
import { MatSlideToggleModule } from "@angular/material/slide-toggle";
import { MatTableModule } from '@angular/material/table';
import { ColumnAreaTextComponent } from './columns/column-area-text/column-area-text.component';
import { ColumnIconActionComponent } from './columns/column-icon-action/column-icon-action.component';
import { ColumnNumberComponent } from './columns/column-number/column-number.component';
import { ColumnSelectComponent } from './columns/column-select/column-select.component';
import { ColumnTextComponent } from './columns/column-text/column-text.component';
import { ColumnTwoCasesComponent } from "./columns/column-two-cases/column-two-cases.component";
import { TableComponent } from './table/table.component';
import { TooltipComponent } from "./tooltip/tooltip.component";
import { DragDropModule } from '@angular/cdk/drag-drop';
import * as i0 from "@angular/core";
export class TableMaterialModule {
}
TableMaterialModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.2", ngImport: i0, type: TableMaterialModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TableMaterialModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.2", ngImport: i0, type: TableMaterialModule, declarations: [TableComponent,
        ColumnTextComponent,
        ColumnAreaTextComponent,
        ColumnIconActionComponent,
        TooltipComponent,
        ColumnNumberComponent,
        ColumnSelectComponent,
        ColumnTwoCasesComponent], imports: [CommonModule,
        FormsModule,
        MatButtonModule,
        MatFormFieldModule,
        MatIconModule,
        MatInputModule,
        MatTableModule,
        MatSelectModule,
        MatPaginatorModule,
        LayoutModule,
        MatTableModule,
        MatSlideToggleModule,
        DragDropModule], exports: [TableComponent,
        ColumnTextComponent,
        ColumnAreaTextComponent,
        ColumnIconActionComponent,
        TooltipComponent,
        ColumnNumberComponent,
        ColumnSelectComponent,
        ColumnTwoCasesComponent] });
TableMaterialModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.2", ngImport: i0, type: TableMaterialModule, imports: [CommonModule,
        FormsModule,
        MatButtonModule,
        MatFormFieldModule,
        MatIconModule,
        MatInputModule,
        MatTableModule,
        MatSelectModule,
        MatPaginatorModule,
        LayoutModule,
        MatTableModule,
        MatSlideToggleModule,
        DragDropModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.2", ngImport: i0, type: TableMaterialModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        TableComponent,
                        ColumnTextComponent,
                        ColumnAreaTextComponent,
                        ColumnIconActionComponent,
                        TooltipComponent,
                        ColumnNumberComponent,
                        ColumnSelectComponent,
                        ColumnTwoCasesComponent,
                    ],
                    imports: [
                        CommonModule,
                        FormsModule,
                        MatButtonModule,
                        MatFormFieldModule,
                        MatIconModule,
                        MatInputModule,
                        MatTableModule,
                        MatSelectModule,
                        MatPaginatorModule,
                        LayoutModule,
                        MatTableModule,
                        MatSlideToggleModule,
                        DragDropModule
                    ],
                    exports: [
                        TableComponent,
                        ColumnTextComponent,
                        ColumnAreaTextComponent,
                        ColumnIconActionComponent,
                        TooltipComponent,
                        ColumnNumberComponent,
                        ColumnSelectComponent,
                        ColumnTwoCasesComponent,
                    ],
                }]
        }] });
